publishing/publish.sh

```diff
@@ -0,0 +1,142 @@
+#!/bin/bash
+
+set -o errexit
+set -o pipefail
+set -o nounset
+
+main() {
+
+  checkScriptDependencies
+  checkParameters
+
+  echo ""
+  echo "------------------------------------------"
+  echo "==> Attempting to publish tag: ${ENV_TAG_NAME}"
+  echo "------------------------------------------"
+  echo ""
+
+  prepareSourceCode
+
+  checkTag
+
+  executeGradlePublish
+
+  echo ""
+  echo "------------------------------------------"
+  echo "==> Tag ${ENV_TAG_NAME} published successfuly."
+  echo "------------------------------------------"
+  echo ""
+}
+
+checkScriptDependencies() {
+    if ! command -v git > /dev/null; then
+        echo "ERROR: git is not installed."
+        exit 1
+    fi
+
+    if ! command -v java > /dev/null; then
+        echo "ERROR: java is not installed."
+        exit 1
+    fi
+}
+
+checkParameters() {
+    if [[ ! -f "${ENV_GRADLE_PROPS_DIR}/gradle.properties" ]]; then
+        echo "PUBLISHING ERROR: '${ENV_GRADLE_PROPS_DIR}/gradle.properties' file does not exist."
+        help
+        exit 1
+    fi
+
+    if [[ ! -f ${ENV_SIGNING_SECRET_KEY_FILE} ]]; then
+        echo "PUBLISHING ERROR: Signing secret file '${ENV_SIGNING_SECRET_KEY_FILE}' does not exist."
+        help
+        exit 1
+    fi
+
+    if [[ -z ${ENV_TAG_NAME:-} ]]; then
+        echo "PUBLISHING ERROR: Missing mandatory environment variable: ENV_TAG_NAME"
+        help
+        exit 1
+    fi
+}
+
+prepareSourceCode() {
+  git clone --branch "$ENV_TAG_NAME"  https://github.com/nixer-io/nixer-spring-plugin.git
+
+  cd nixer-spring-plugin
+
+  echo ""
+  echo "--> Checked out repository: $(git config --get remote.origin.url)"
+  echo ""
+}
+
+checkTag() {
+  CURRENT_TAG=$(git describe --tags)
+
+  if [[ "$CURRENT_TAG" != "$ENV_TAG_NAME" ]]; then
```

Comment:
Add comment or modify this condition to handle more than one tag on a commit. (which might be a correct scenario)

Reply:
In case of multiple tags on a commit `git describe` shows only the one created most recently. I extended the error message to explain this. I also changed the check so it requires only annotated tags and the current commit matching the tag exactly.